File: RestaurantHubBackend/src/main/java/com/restauranthub/service/OrderService.java
package com.restauranthub.service;

import java.util.List;
import java.util.Optional;

import com.restauranthub.model.Order;

public interface OrderService {

    Order placeOrder(Order order);

    List<Order> getAllOrders();

    Optional<Order> getOrderById(Long id);

    List<Order> getOrdersByEmail(String email);

    List<Order> getOrdersByStatus(String status);

    Order updateOrder(Long id, Order order);

    String deleteOrder(Long id);

}

File: frontend/src/services/orderService.js
import API from "./api";

// Place Order
export const placeOrder = (order) => {

    return API.post("/orders", order);

};

// Get All Orders
export const getOrders = () => {

    return API.get("/orders");

};

// Get Order By ID
export const getOrderById = (id) => {

    return API.get(`/orders/${id}`);

};

// Get Orders By Email
export const getOrdersByEmail = (email) => {

    return API.get(`/orders/email/${email}`);

};

// Update Order
export const updateOrder = (id, order) => {

    return API.put(`/orders/${id}`, order);

};

// Approve / Change Order Status
export const approveOrder = (id, order) => {

    return API.put(`/orders/${id}`, order);

};

// Delete Order
export const deleteOrder = (id) => {

    return API.delete(`/orders/${id}`);

};